src/modes/edit_message.cc

```diff
@@ -134,27 +135,41 @@ namespace Astroid {
     id = edit_id++;
 
     ustring _mid = "";
-    msg_time = time(0);
+
+    /* defaults */
+    accounts = astroid->accounts;
 
 # ifndef DISABLE_PLUGINS
     if (!astroid->plugin_manager->astroid_extension->generate_mid (_mid)) {
 # endif
 
-      char _hostname[1024];
-      _hostname[1023] = 0;
-      gethostname (_hostname, 1023);
-
-      char _domainname[1024];
-      _domainname[1023] = 0;
-      if (getdomainname (_domainname, 1023) < 0) {
-        *_domainname = '\0';
-      }
-
+      //
+      // generate left-hand side of the message id
+      //
+      // get value for right-hand side from config file
       ustring hostname = astroid->config ().get <string> ("mail.message_id_fqdn");
       UstringUtils::trim (hostname);
+      // if hostname contains a "%1", insert the FQDN of the default
+      // account's email address
+      if (hostname.find ("%1", 0) != std::string::npos) {
+        Account &a = (accounts->accounts)[accounts->default_account];
+        hostname = ustring::compose (hostname,
+                                     boost::replace_all_copy (string(fqdn_of_address(a.email)), "%", "%%"));
```

Comment:
The extra replace_all_copy is here to safeguard of messages where a.email might contain `%`. This might still be good to have.